pkg/controller/flinkapplication/flink_state_machine.go

```diff
@@ -312,12 +319,22 @@ func (s *FlinkStateMachine) handleApplicationSavepointing(ctx context.Context, a
 
 	// check the savepoints in progress
 	savepointStatusResponse, err := s.flinkController.GetSavepointStatus(ctx, application, application.Status.DeployHash)
-	if err != nil {
+	// Here shouldRollback() is used as a proxy to identify if we have exhausted all retries trying to GetSavepointStatus
+	// The application is NOT actually rolled back  because we're in a spot where there may be no application running and we don't
+	// have information on the last successful savepoint.
+	// In the event that rollback evaluates to True, we don't return immediately but allow for the remaining steps to proceed as normal.
+	rollback, reason := s.shouldRollback(ctx, application)
+
+	if err != nil && !rollback {
 		return statusUnchanged, err
 	}
 
+	s.flinkController.LogEvent(ctx, application, corev1.EventTypeWarning, "SavepointStatusCheckFailed",
```

Comment:
It looks like we'll print this out every time we savepoint, regardless of whether this failed

Reply:
My bad. Yes. Fixed.